### PR TITLE
Added REVIEW Week 7 content

### PR DESCRIPTION
**Pull request by remarcmij**
Discussion of map, filter and reduce.

### CODE REVIEW

File: Week7/REVIEW.md [hunk 1]
```diff
@@ -1,9 +1,196 @@
...
+console.log(groupedNames);
 ```
-This review covers:
- • Git Workflow
- • Map, 
- • Reduce 
- • Filter
-```
\ No newline at end of file
+
```
Comment (mkruijt) [inline Week7/REVIEW.md]: Hi Jim, can we keep this list at the top?

File: Week7/REVIEW.md [hunk 2]
```diff
@@ -1,9 +1,196 @@
 # REVIEW JavaScript week 7
 
+## Git Workflow
+
+To be provided
```
Comment (mkruijt) [inline Week7/REVIEW.md]: here we can add a link to: https://github.com/HackYourFuture/Git/blob/master/Lecture-3.md

Comment (mkruijt) [reply Week7/REVIEW.md]: the workflow is explained there

File: Week7/REVIEW.md [hunk 3]
```diff
@@ -1,9 +1,196 @@
...
+```js
+const numbers = [3, 5, 2, 7];
+const squares = numbers.map(num => num * num);
+console.log(squares); // -> [9, 25, 4, 49]
+```
+
+### Array#filter
+
+The **filter()** method returns a new array with all elements that pass the test implemented by a user-supplied (predicate\*) function. 
+
```
Comment (mkruijt) [inline Week7/REVIEW.md:86]: maybe you can describe a bit more what the function below does, (in a simpler way)

File: Week7/REVIEW.md [hunk 4]
```diff
@@ -1,9 +1,196 @@
...
+To be provided
+
+## Map, filter, reduce 
+
+The array methods **map()**, **filter()** and **reduce()** are best understood by looking at how they could be implemented if we were to write them ourselves. In the next few sections we will present simplified versions of the native implementations. We have prefixed the method names with `my` to distinguish them from the built-in versions.
+
+### Array#map\*
+
+The **map** method returns a new array where each element of the subject array is transformed by a user-supplied transformation (= _mapping_) function. 
+
```
Comment (mkruijt) [inline Week7/REVIEW.md:24]: maybe you can describe a bit more what the function below does, (in a simpler way) and with a real life example, something people can actually relate to

File: Week7/REVIEW.md [hunk 5]
```diff
@@ -1,9 +1,196 @@
...
+As is usual in JavaScript you do not necessarily have to use all the parameters that were passed to the `mapFn` function. In fact, in many cases you will only need the first argument (the current array element).
+
+In the example below we will use the Array#map method to compute the squares of an array of numbers.
+
+```js
+const numbers = [3, 5, 2, 7];
+const squares = numbers.map(num => num * num);
+console.log(squares); // -> [9, 25, 4, 49]
+```
```
Comment (mkruijt) [inline Week7/REVIEW.md:55]: I like that you say, "here we do the same using the map/filter/reduce method"

File: Week7/REVIEW.md [hunk 6]
```diff
@@ -1,9 +1,196 @@
...
+
+### Array#reduce
+
+Of the three methods **map**, **filter** and **reduce**, the **reduce** method presents the most difficulty for new learners. The _Mozilla Developer Network_ (MDN) web site gives the following definition:
+
+> The reduce() method applies a function against an accumulator and each element in the array (from left to right) to reduce it to a single value\*\*.
+
+<small>\*\*Although reference is made to a 'single value', this single value may well be an array or an object, as you will see later in the examples below.</small>
+
```
Comment (mkruijt) [inline Week7/REVIEW.md:118]: same comment as above. It would be nice if the coding examples have a bit more context. explanation

**Review by mkruijt — CHANGES_REQUESTED** [review]
Hi Jim, looks really good. For me the coding examples work better if you have a bit more context, something like, (step 6) https://github.com/HackYourFuture/JavaScript/blob/master/Week7/MAKEME.md . Also can you make sure that at the top there is a list that tells what is in the document?

File: Week7/REVIEW.md (hunk 5 above)
Comment (remarcmij) [reply Week7/REVIEW.md:55]: Not sure what you want me to do here...

Comment (mkruijt) [reply Week7/REVIEW.md:55]: nothing, it was a compliment!

**Review by mkruijt — APPROVED** [review]
Thank you for the changes!

**Pull request merged**